client/src/app/pages/applications/components/application-detail-drawer/components/download-button.tsx

```diff
@@ -1,32 +1,26 @@
-import React, { useState } from "react";
-import { Alert, Button } from "@patternfly/react-core";
+import React from "react";
+import { Alert, Button, Spinner } from "@patternfly/react-core";
 import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
-import { Application } from "@app/api/models";
-import { Spinner } from "@patternfly/react-core";
-import { useDownloadStaticReport } from "@app/queries/download";
+import { saveAs } from "file-saver";
 
-export enum MimeType {
-  TAR = "tar",
-  YAML = "yaml",
+import { MimeType } from "@app/api/models";
+import { useFetchStaticReport } from "@app/queries/applications";
+
+interface IDownloadButtonProps {
+  id: number;
+  mimeType: MimeType;
 }
-function DownloadButton({
-  application,
+
+export const DownloadButton: React.FC<IDownloadButtonProps> = ({
+  id,
   mimeType,
-}: {
-  application: Application;
-  mimeType: MimeType;
-}) {
-  const {
-    mutate: downloadFile,
-    isLoading,
-    isError,
-  } = useDownloadStaticReport();
+}) => {
+  const { report, isLoading, isError } = useFetchStaticReport(id, mimeType);
```

Comment:
Looks like this is causing the report to be fetched on load rather than on click when the button is implemented in the drawer. This causes the loading state to be triggered and the request ultimately fails due to nothing being returned from the downloadApplicationAnalysisReport function.